project_name: nomad
run_name: nomad

# load_run: nomad_run # comment out if training from scratch

# diffusion model params
num_diffusion_iters: 10

# action output params
len_traj_pred: 8
learn_angle: False

# context
context_type: temporal
context_size: 3
alpha: 1e-4
beta: 0.1

# normalization for the action space
normalize: True

# dataset specific parameters
image_size: [96, 96] # width, height
datasets:
  training_data:
    end_slack: 3
    goals_per_obs: 1
    negative_mining: True
    
# training setup
use_wandb: True # set to false if you don't want to log to wandb
train: True
batch_size: 32
eval_batch_size: 32
epochs: 10
gpu_ids: [0]
num_workers: 2
lr: 1e-4
optimizer: adamw
clipping: False
max_norm: 1.
scheduler: "cosine"
warmup: True
warmup_epochs: 4
cyclic_period: 10
plateau_patience: 3
plateau_factor: 0.5
seed: 0
train_subset: 1.

# model params
model_type: nomad
vision_encoder: nomad_vint
encoding_size: 256
obs_encoder: efficientnet-b0
attn_unet: False
cond_predict_scale: False
mha_num_attention_heads: 4
mha_num_attention_layers: 4
mha_ff_dim_factor: 4
down_dims: [64, 128, 256]
obs_encoding_size: 1024
goal_encoding_size: 1024

# mask
goal_mask_prob: 0.5

obs_type: image
goal_type: image

# distance bounds for distance and action and distance predictions 
distance:
  min_dist_cat: 0
  max_dist_cat: 20
action:
  min_dist_cat: 3
  max_dist_cat: 20
close_far_threshold: 10 # distance threshold used to separate the close and the far subgoals that are sampled per datapoint

# logging stuff
## =0 turns off
print_log_freq: 100 # in iterations
image_log_freq: 1000 # in iterations
num_images_log: 8
pairwise_test_freq: 0 # in epochs
eval_fraction: 0.25
wandb_log_freq: 10 # in iterations
eval_freq: 1 # in epochs
